### PR TITLE
[GH-16126] - Migrate 'components/integrations/edit_command' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by lucievr**
#### Summary
- Migrated `components/integrations/edit_command` module and test files to TypeScript
- Renamed files to have TypeScript extensions and updated incorrect imports
- Created a test helper method for mocking a command in `utils/test_helper.ts` which is then used in `components/integrations/edit_command` test file
- Check-types check returned no errors

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/16126

### CODE REVIEW

File: components/integrations/edit_command/__snapshots__/edit_command.test.tsx.snap [hunk 1]
```diff
@@ -48,7 +48,7 @@ exports[`components/integrations/EditCommand should match snapshot 1`] = `
       "auto_complete": true,
       "auto_complete_desc": "auto_complete_desc",
       "auto_complete_hint": "auto_complete_hint",
-      "create_at": "1499722850203",
+      "create_at": 1499722850203,
```
Comment (lucievr) [inline components/integrations/edit_command/__snapshots__/edit_command.test.tsx.snap:51]: Assumed this timestamp was supposed to be a number as defined here https://github.com/mattermost/mattermost-redux/blob/master/src/types/integrations.ts

File: components/integrations/edit_command/index.ts [hunk 2]
```diff
@@ -2,14 +2,26 @@
...
-import EditCommand from './edit_command.jsx';
+import EditCommand from './edit_command';
 
-function mapStateToProps(state, ownProps) {
+type Props = {
+    location: Location;
+}
+
+type Actions = {
+    getCustomTeamCommands: (teamId: string) => Promise<any>;
```
Comment (devinbinnie) [inline components/integrations/edit_command/index.ts:20]: This returns an array of `Command` objects, so we don't have to use `any` here.

**Review by devinbinnie — CHANGES_REQUESTED** [review]
Thanks for the PR @lucievr! Looks good, I just have one change to request.

**Review by hmhealey — APPROVED** [review]
Looks good except for the other change requested.

Thanks for the help on this migration, @lucievr!

**Comment by lucievr** [comment]
Thank you @devinbinnie and @hmhealey for the review and comments! I've made the requested change and tweaked the test file a bit too as it was otherwise throwing errors with the new change. Hope all is good now.

**Review by devinbinnie — APPROVED** [review]

**Comment by devinbinnie** [comment]
LGTM! Thanks @lucievr!

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Regression tested creating updating and deleting of custom slash commands
- Tested on desktop and browser
- I did see an intermittent JS error but it was unrelated to this PR and will be addressed by https://mattermost.atlassian.net/browse/MM-31515 
- No other functional or JS issues found.

LGTM!
No new release tests needed to cover this change.
Thanks @lucievr for this PR!

**Pull request merged**